agent/taskresource/volume/dockervolume_test.go

```diff
@@ -113,17 +115,66 @@ func TestCleanupError(t *testing.T) {
 	autoprovision := false
 	driver := "driver"
 
-	mockClient.EXPECT().RemoveVolume(gomock.Any(), name, dockerapi.RemoveVolumeTimeout).Return(errors.New("some error"))
-
 	ctx, cancel := context.WithCancel(context.TODO())
 	defer cancel()
-	volume := NewVolumeResource(name, scope, autoprovision, driver, nil, nil, mockClient, ctx)
+	volume, _ := NewVolumeResource(ctx, name, name, scope, autoprovision, driver, nil, nil, mockClient)
 	err := volume.Cleanup()
-	assert.NotNil(t, err)
+	assert.Nil(t, err)
+}
+
+func TestApplyTransitionForTaskScopeVolume(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockClient := mock_dockerapi.NewMockDockerClient(ctrl)
+
+	name := "volumeName"
+	scope := "task"
+	autoprovision := false
+	driver := "driver"
+	driverOptions := map[string]string{}
+	labels := map[string]string{}
+	mountPoint := "some/mount/point"
+
+	gomock.InOrder(
+		mockClient.EXPECT().CreateVolume(gomock.Any(), name, driver, driverOptions, labels, dockerapi.CreateVolumeTimeout).Times(1).Return(
+			dockerapi.VolumeResponse{
+				DockerVolume: &docker.Volume{Name: name, Driver: driver, Mountpoint: mountPoint, Labels: nil},
+				Error:        nil,
+			}),
+	)
+
+	volume, _ := NewVolumeResource(nil, name, name, scope, autoprovision, driver, driverOptions, labels, mockClient)
+	volume.ApplyTransition(resourcestatus.ResourceStatus(VolumeCreated))
+}
+
+func TestApplyTransitionForSharedScopeVolume(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockClient := mock_dockerapi.NewMockDockerClient(ctrl)
+
+	name := "volumeName"
+	scope := "shared"
```

Comment:
We do not cleanup shared volumes. Do we still expect `RemoveVolume` call here?

Reply:
No, as you can see the times is zero. but I think we can just remove that `EXPECT`.